onnxruntime/core/providers/coreml/builders/model_builder.cc

```diff
@@ -143,18 +181,15 @@ void CopyOnnxTensorToCoreMLTensor(const ONNX_NAMESPACE::TensorProto& tensor_prot
       break;
     }
     case ONNX_NAMESPACE::TensorProto_DataType_INT64: {
-      // enable when this is proven to not be the case
-      ORT_THROW(
-          "INT64 is unexpected as CoreML uses 32-bit int for indices. "
-          "Most likely an initializer that should have been skipped was not.");
-      //// from: int64_data/raw, to: longints
-      // if (has_raw_data) {
-      //   CopyRawDataToRepeatedField<int64_t>(tensor_proto, *tensor_value.mutable_longints()->mutable_values());
+      // from: int64_data/raw, to: ints (use narrow to convert to int32)
+      // CoreML uses int32
```

Comment:
is this something we want to do for all int64 inputs?

Reply:
Ideally, yeah, every CoreML op should have their int64 inputs converted to int32 and their int64 outputs converted to int32. CoreML doesn't support int64 for their ops.